piv/key.go

```diff
@@ -102,6 +103,11 @@ type Attestation struct {
 	PINPolicy PINPolicy
 	// TouchPolicy set on the slot.
 	TouchPolicy TouchPolicy
+
+	// Slot is the two-letter name of the slot (9a, 9c, etc.) the attested key resides in.
+	// See https://developers.yubico.com/PIV/Introduction/Certificate_slots.html for more
+	// information.
+	Slot string
```

Comment:
We already have a slot object. Would you want to make this an actual type, rather than just a string?

```
// Inferred Slot based on the common name of the certificate. If the slot can't be determined, this
// field will be equivalent to an empty Slot struct.
Slot Slot
```

We could also give Slot a String() function to make printing it easier

```
func (s *Slot) String() string {
   return strconv.FormatUint(uint64(s.Key), 16)
}
```

Parsing logic then becomes something like:

```
const yubikeySubjectCNPrefix = "YubiKey PIV Attestation "

func parseSlot(commonName string) (Slot, bool) {
    if strings.HasPrefix(commonName, yubikeySubjectCNPrefix) {
        return Slot{}, false
    }
    slotName := strings.TrimPrefix(commonName, yubikeySubjectCNPrefix(
    key, err := strconv.ParseUint(slotName, 16, 32)
    if err != nil {
        return Slot{}, false
    }
    switch key {
    case SlotAuthentication.Key:
        return SlotAuthentication, true
    case SlotSignature.Key:
        return SlotSignature, true
    case SlotCardAuthentication.Key:
        return SlotCardAuthentication, true
    case SlotKeyManagement.Key:
        return SlotKeyManagement, true
    }
    return RetiredKeyManagementSlot(key)
}
```

That also makes unit testing parseSlot possible.

Reply:
This makes sense, it will make it easier for projects using the library to compare the slots than relying on string comparisons.